release/ray_release/tests/test_test.py

```diff
@@ -5,22 +5,48 @@
 from ray_release.test import Test
 
 
+def _stub_test(val: dict = {}) -> Test:
+    test = Test(
+        {
+            "name": "test",
+            "cluster": {},
+        }
+    )
+    test.update(val)
+    return test
+
+
 def test_get_python_version():
-    assert Test().get_python_version() == "3.7"
-    assert Test({"python": "3.8"}).get_python_version() == "3.8"
+    assert _stub_test().get_python_version() == "3.7"
+    assert _stub_test({"python": "3.8"}).get_python_version() == "3.8"
 
 
 def test_get_ray_image():
     os.environ.pop("BUILDKITE_COMMIT", None)
-    assert Test({"python": "3.8"}).get_ray_image() == "rayproject/ray:nightly-py38"
+    assert (
+        _stub_test({"python": "3.8"}).get_ray_image() == "rayproject/ray:nightly-py38"
+    )
+    assert (
+        _stub_test(
+            {
+                "python": "3.8",
+                "cluster": {
+                    "byod": {
+                        "type": "gpu",
+                    }
+                },
+            }
+        ).get_ray_image()
+        == "rayproject/ray-ml:nightly-py38-gpu"
```

Comment:
why byod can still get a ray image?

Reply:
you mean nightly image? It will get a proper ray-ml:hash-py38-gpu when the BUILDKITE_COMMIT is supplied

Reply:
should we test that (with a commit) instead?

I guess this is for the base image for building the byod image? I am a bit confused where byod returns a nightly ray image.

Reply:
I have a test with a commit in subsequent pr, the current code in this PR just support nightly for now